### DISCUSSION TITLE
Default Settings

### DISCUSSION THREAD

**Discussion by madprops** [madprops/grasshopper#142] [General]
Here's a space to discuss setting defaults that you think should be changed. 

There's lots of settings so it's hard to find the perfect configuration.

I think menus like in hover button and close button can be improved, since I just added some demo commands expecting users to know they can modify them.

**Comment by N3C2L** [comment]
I find almost all default settings very well done. You can immediately find your way around and adjust so much according to your taste. 👍

In the default menus of the hover button and close button, commands could perhaps be added that lead directly to the settings of these two. That means adding the command "Settings: Close" to the default close menu and the command "Settings: Hover" to the default hover menu. This would certainly be helpful for newcomers. 

With the Extra Menu (context menu), however, when right-clicking, you don't know which of these commands opens a submenu. For example, when I click on "filter", a submenu opens. But it's not apparent that "filter" opens a submenu. This means that perhaps an arrow icon should be displayed to the right of "filter" to show that a different submenu opens when you click on it. 

Commands from these submenus could also be distributed into the menus of the hover menu and close menu. Personally, I have many custom colors and custom URLs in the hover menus and close menus, but also combos.

**Comment by madprops** [comment]
Noted. Gonna add the hover and close settings shortcuts for now.

**Comment by N3C2L** [comment]
Maybe the taglist should be activated by default.
Show taglist: check. 
Because a user wants to know where their tag is (after he tagged a tab), what they can do with it (change, filter, etc. all as quickly as possible with the taglist).

**Reply by madprops** [reply]
Ok, I guess that makes tags more useful. I didn't want to clutter the UI too much, but if people are gonna use tags then the taglist should be enabled.

**Reply by madprops** [reply]
I'm going to change the automatic tag picker, as I don't like it.

I want to add them more manually and use the icon when needed.

To use the list click the icon on the right manually.

**Reply by N3C2L** [reply]
Ok, sounds useful! But I'd appreciate a setting to activate the actual automatic tag picker as I use it at this time. So people can choose which one they like.

**Reply by madprops** [reply]
Ok added `Auto Tag Picker` in `More`.

**Reply by N3C2L** [reply]
Thanks!

**Comment by N3C2L** [comment]
When someone installs GH for the first time and tests the bookmark mode function by performing a search, they may find that not all of their folders or bookmarks are being searched. This is because the default values are currently set as follows:

Max bookmark folders to fetch: 100
Deep max search: 5000

However, how would the user know that they need to increase these values first to ensure their bookmark search covers everything? This means a user would need to navigate to the bookmark settings and filter settings, and increase the values to match their current total available bookmarks (I have 99999 as values for both the deep max search and the max bookmark folders to fetch cmd - just to be sure I don't need to alter this value anytime) to truly search all the bookmarks. Since you have now implemented the bookmark server, the search is so fast that I would set both "auto deep bookmarks" and "auto deep history" to be enabled by default. Plus the minimum bookmarks folders to fetch and the deep max search shouldn't be limited by default.
And how should a new user know, that he must perform the deep search cmd if a bookmarks search doesn't find a result? There isn't a popup that says: "No results: try a deep bookmark search that fetches all your bookmarks".

**Comment by madprops** [comment]
I added some messages when search returns nothing.

Also added more settings to tab box. Default now shows a scrollbar.

I'm gonna just make scrollbars on by default, to avoid annoying users too much.

It can always be disabled in `Show`.
